distribution/lib/Standard/Visualization/0.0.0-dev/src/Text.enso

```diff
@@ -0,0 +1,51 @@
+from Standard.Base import all
+
+from Standard.Base.Data.Text.Extensions import slice_text
+
+## Return a sub-window of a string. The window is defined by line/chunk coordinates. The siz of
+   a chunk is defined by `chunk_width`. The output is formatted as a message that can be sent to
+   the IDE's lazy text visualisation.
+get_lazy_visualisation_text_window text pos size chunk_width =
+    get_text_chunk = get_item_from text chunk_width
+    pos_x = Math.max (pos.at 0) 0
+    pos_y = Math.max (pos.at 1) 0
+    size_x = size.at 0
+    size_y = size.at 1
+    x_range = pos_x.up_to (pos_x + size_x)
+    y_range = pos_y.up_to (pos_y + size_y)
+    coordinates = x_range.map (x -> y_range.map (y -> [x,y])) . flatten
+    chunks = coordinates.map (ix -> format_chunk ix (get_text_chunk ix))
+    data_string = format_chunks chunks
+    lines = text.lines.length
+    active_lines = y_range.map (line_ix -> text.lines.at line_ix)
+    max_line_length = (active_lines.map (line -> line.length)).fold 0 (l -> r -> Math.max l r)
+    make_grid_visualisation_response data_string lines max_line_length
+
+## Format a chunk of text and meta information for the lazy visualisation.
+make_grid_visualisation_response chunks lines max_line_length =
+    (Json.from_pairs [["chunks", chunks], ["lines", lines], ["max_line_length", max_line_length]]).to_text
+
+## Return a chunk of text from a string. The chunk is defined by a its size and a line/chunk index
+   coordinate.
+get_item_from text chunk_size index =
+    line_ix = index.at 1
+    if line_ix > text.lines.length then "" else
```

Comment:
This was intentional because all the visualization will just want to know that there is nothing to display. We could try to omit the value for the requested cell, but then it will be harder to differentiate between "I have requested this and know that there is nothing to display" and "I do not know the value to display". 

I guess we could also return `null` (in JSON) and this would probably be the nicer design for potential optimizations on the IDE side. So I'm changing it to that.